api/krusty/basic_io_test.go

```diff
@@ -80,3 +81,35 @@ spec:
   clusterIP: None
 `)
 }
+
+//test for https://github.com/kubernetes-sigs/kustomize/issues/3812#issuecomment-862339267
+func TestBasicIO3812(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
+	th.WriteK(".", `
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - service.yaml
+`)
+
+	th.WriteF("service.yaml", `
+apiVersion: v1
+kind: Service
+metadata:
+  name: kapacitor
+  labels:
+    app.kubernetes.io/name: tick-kapacitor
+spec:
+  selector:
+    app.kubernetes.io/name: tick-kapacitor
+    - name: http
+      port: 9092
+      protocol: TCP
+  type: ClusterIP
+`)
+
+	err := th.RunWithErr(".", th.MakeDefaultOptions())
+	if !kusterr.IsMalformedYAMLError(err) {
+		t.Fatalf("unexpected error: %q", err)
+	}
```

Comment:
Please also add a check for the error contents, e.g. `assert.Equal(t, err.Error(), expectedErr)`